src/Features/Core/Portable/PullMemberUp/AbstractMembersPullerService.cs

```diff
@@ -300,10 +310,58 @@ private static ISymbol GetSymbolsToPullUp(MemberAnalysisResult analysisResult)
                 newDestination = destinationEditor.Generator.WithModifiers(newDestination, modifiers);
             }
 
-            destinationEditor.ReplaceNode(destinationSyntaxNode, (syntaxNode, generator) => newDestination);
+            // add imports by moving all source imports to destination container, then taking out unneccessary
+            // imports that we just added.
+            var syntaxFactsService = destinationEditor.OriginalDocument.Project.LanguageServices.GetRequiredService<ISyntaxFactsService>();
+            var destinationTrivia = GetLeadingTriviaBeforeFirstMember(destinationEditor.OriginalRoot, syntaxFactsService);
+            var addImportsService = destinationEditor.OriginalDocument.Project.LanguageServices.GetRequiredService<IAddImportsService>();
+            var semanticModel = await destinationEditor.GetChangedDocument().GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            destinationEditor.ReplaceNode(destinationEditor.OriginalRoot, (node, generator) => addImportsService.AddImports(
+                semanticModel.Compilation,
+                node,
+                node.GetCurrentNode(destinationSyntaxNode),
+                sourceImports,
+                generator,
+                options.PlaceSystemNamespaceFirst,
+                destinationEditor.OriginalDocument.CanAddImportsInHiddenRegions(),
+                cancellationToken));
+
+            destinationEditor.ReplaceNode(destinationSyntaxNode, newDestination);
+
+            var removeImportsService = destinationEditor.OriginalDocument.Project.LanguageServices.GetRequiredService<IRemoveUnnecessaryImportsService>();
+            var destinationDocument = await removeImportsService.RemoveUnnecessaryImportsAsync(
+                destinationEditor.GetChangedDocument(),
+                node => node.GetAnnotations(Kind).Any(),
+                cancellationToken)
+                .ConfigureAwait(false);
+
+            var destinationRoot = AddLeadingTriviaBeforeFirstMember(
+                await destinationDocument.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false),
+                syntaxFactsService,
+                destinationTrivia);
+            destinationEditor.ReplaceNode(destinationEditor.OriginalRoot, destinationRoot);
+
             return solutionEditor.GetChangedSolution();
         }
 
+        // In the case where we have imports from the source that get moved up, but none are necessary
+        // and are all removed, the adding/removing can destroy or copy leading trivia, including comments.
+        // So, we grab the leading trivia before we do import operations, and add it back in after.
+        private static SyntaxTriviaList GetLeadingTriviaBeforeFirstMember(SyntaxNode root, ISyntaxFactsService factsService)
```

Comment:
```suggestion
        private static SyntaxTriviaList GetLeadingTriviaBeforeFirstMember(SyntaxNode root, ISyntaxFactsService syntaxFacts)
```

we don't ever really say 'factsService' anywhere else.